Sources/Operators/RetryWhen.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  RetryWhen.swift
+//  CombineExt
+//
+//  Created by Daniel Tartaglia on 3/21/20.
+//
+
+#if canImport(Combine)
+import Combine
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+public extension Publisher {
+
+    /// Repeats the source publisher on error when the notifier emits a next value. If the source publisher errors and the notifier completes, it will complete the source sequence.
+    ///
+    /// - Parameter notificationHandler: A handler that is passed a publisher of errors raised by the source publisher and returns a publisher that either continues, completes or errors. This behavior is then applied to the source publisher.
+    /// - Returns: A publisher producing the elements of the given sequence repeatedly until it terminates successfully or is notified to error or complete.
+    func retryWhen<Trigger>(_ notificationHandler: @escaping (AnyPublisher<Self.Failure, Never>) -> Trigger)
+        -> Publishers.RetryWhen<Self, Trigger, Output, Failure> where Trigger: Publisher {
+            .init(upstream: self, notificationHandler: notificationHandler)
+    }
+}
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+public extension Publishers {
+    class RetryWhen<Upstream, Trigger, Output, Failure>: Publisher where Upstream: Publisher, Upstream.Output == Output, Upstream.Failure == Failure, Trigger: Publisher {
+        
+        typealias Handler = (AnyPublisher<Upstream.Failure, Never>) -> Trigger
+        
+        private let upstream: Upstream
+        private let handler: Handler
+        
+        init(upstream: Upstream, notificationHandler: @escaping Handler) {
+            self.upstream = upstream
+            self.handler = notificationHandler
+        }
+        
+        public func receive<S>(subscriber: S) where S: Subscriber, Failure == S.Failure, Output == S.Input {
+            subscriber.receive(subscription: Subscription(upstream: upstream, downstream: subscriber, handler: handler))
+        }
+    }
+}
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+extension Publishers.RetryWhen {
+    class Subscription<Downstream>: Combine.Subscription where Downstream: Subscriber, Downstream.Input == Upstream.Output, Downstream.Failure == Upstream.Failure {
+        
+        private let upstream: Upstream
+        private let downstream: Downstream
+        private let errorSubject = PassthroughSubject<Upstream.Failure, Never>()
+        private var sink: Sink<Upstream, Downstream>?
+        private var cancellable: AnyCancellable?
+        
+        init(upstream: Upstream, downstream: Downstream, handler: @escaping (AnyPublisher<Upstream.Failure, Never>) -> Trigger) {
```

Comment:
split arguments into their own lines here
```suggestion
        init(
            upstream: Upstream,
            downstream: Downstream,
            handler: @escaping (AnyPublisher<Upstream.Failure, Never>) -> Trigger
        ) {
```